packages/flutter_tools/test/general.shard/resident_web_runner_test.dart

```diff
@@ -56,6 +56,12 @@ const List<VmServiceExpectation> kAttachIsolateExpectations = <VmServiceExpectat
       'streamId': 'Isolate'
     }
```

Comment:
I would recommend unit tests to validate:

Extension event for non-structured errors is ignored:
Extension event for invalid json is ignored:
Extension event for valid structured errors is OK

Reply:
The first and last situations are tested, but I'm not sure how to test invalid JSON since the JSON seems to be formed (and presumably checked) when the event is created. Would I need to test with different setup or is this out of scope?

Reply:
I wouldn't worry about invalid JSON then, perhaps just json without a renderedErrorText field?